jazz_events/test/test.js

```diff
@@ -37,478 +50,408 @@ describe('Sample', function() {
+            return res;
+        })).to.be.rejectedWith(message);
         AWS.restore("DynamoDB");
-        stub.restore();
-        assert.isTrue(cbCheck);
     });
 
-    it("should attempt to get item from DynamoDB by body prameters if POST method and body parameters are defined", function(){
-        event.method = "POST";
-        var attemptBool = dynamoCheck('getItem', spy);
-        assert.isTrue(attemptBool);
+    it("should indicate error while validating event specific data if event timestamp is invalid", () => {
+        config = configObj(event);
+        event.body.event_timestamp = "xyz";
+        var message = "Invalid EVENT TIMESTAMP: xyz, The format should be YYYY-MM-DDTHH:mm:ss:SSS"
+        var dataObj = {
+            Item: event.body
+        };
+        AWS.mock("DynamoDB", "getItem", (params, callback) => {
+            return callback(null, dataObj);
+        });
+        var validateEventInput = index.validateEventInput(config, event.body);
+        expect(validateEventInput.then((res) => {
+            return res;
+        })).to.be.rejectedWith(message);
+        AWS.restore("DynamoDB");
     });
 
-    it("should attempt to initiate kinesis putRecord if method POST and body parameters are defined",()=>{
-        event.method = "POST";
-        // mock kinesis putRecord with event.body parameters defined
-        AWS.mock('DynamoDB', 'getItem',spy);
-        AWS.mock("Kinesis","putRecord",spy);
-        var callFunction = index.handler(event, context, callback);
-        var bool = spy.called;
-        AWS.restore('Kinesis');
-        AWS.restore('DynamoDB');
-        assert.isTrue(bool);
+    it("should indicate success to get list of events for provided query params with GET method", () => {
+        event.method = "GET";
+        var eventBody = event.body;
+        var dbResult = {
+            Items: [{
+                'SERVICE_CONTEXT': {
+                    S: eventBody.service_context
+                },
+                'EVENT_HANDLER': {
+                    S: eventBody.event_handler
+                },
+                'EVENT_NAME': {
+                    S: eventBody.event_name
+                },
+                'SERVICE_NAME': {
+                    S: eventBody.service_name
+                },
+                'EVENT_STATUS': {
+                    S: eventBody.event_status
+                },
+                'EVENT_TYPE': {
+                    S: eventBody.event_type
+                },
+                'USERNAME': {
+                    S: eventBody.username
+                },
+                'EVENT_TIMESTAMP': {
+                    S: eventBody.event_timestamp
+                }
+            }]
+        }
+        var result = {
+            events: [event.body]
+        }
+        AWS.mock("DynamoDB", "scan", (params, callback) => {
+            return callback(null, dbResult)
+        });
+        index.handler(event, context, (err, res) => {
```

Comment:
Ideally in tests you should have either error(reject) or resolve(success). We should have separate tests to handle it.
Same for other tests

Reply:
Worked on it.